src/main/java/org/opensearch/knn/index/mapper/CompressionLevel.java

```diff
@@ -105,20 +107,29 @@ public static boolean isConfigured(CompressionLevel compressionLevel) {
      *                  or equal to 1000, the default {@link RescoreContext} if greater, or {@code null} if the mode
      *                  is invalid.
      */
-    public RescoreContext getDefaultRescoreContext(Mode mode, int dimension) {
+    public RescoreContext getDefaultRescoreContext(Mode mode, int dimension, Version version) {
+        // TODO move this to separate class called resolver to resolve rescore context
         if (modesForRescore.contains(mode)) {
+            if (this == x4 && version.before(Version.V_3_1_0)) {
+                // For index created before 3.1, context was always null and mode is empty
+                return null;
+            }
             // Adjust RescoreContext based on dimension
             if (dimension <= RescoreContext.DIMENSION_THRESHOLD) {
```

Comment:
Should we skip this logic for 4x which sets oversampling factor to 5.0 if vector dimension is <=1000 as we are seeing good recall with OSF of 1.0 ? 
Or the recall improvement which we have seen in your perf results is because of this logic which sets it to 5.0 as cohere dataset dimension is 768 unless you are overriding the rescore context during query time ?

Reply:
Ack. We can skip this logic.